test/controllers/authorizeService.js

```diff
@@ -0,0 +1,184 @@
+     };
+
+     describe('Get api-key', function () {
+
+          it('should return 401 when the requester is not the WStore', function (done) {
+               var req = {
+                    ip: {
+                         replace: function (expr) { return '130.25.13.12'}
+                    }
+               };
+
+               invalidRequest('getApiKey', req, 401, {error: 'Invalid remote client'}, done);
+          });
+
+          it('should return 400 when the body is empty', function (done) {
+               var req = {
+                    ip: {
+                         replace: function (expr) { return DEFAULT_WSTOREHOST}
+                    }
+               }
+
+               invalidRequest('getApiKey', req, 400, {error: 'Invalid body'}, done);
+          });
+
+          it('should return 400 when the "url" is not defined', function (done) {
+               var req = {
+                    ip: {
+                         replace: function (expr) { return DEFAULT_WSTOREHOST}
+                    },
+                    body: '{}'
+               };
+
+               invalidRequest('getApiKey', req, 400, {error: 'Url missing'}, done);
+          });
+
+          var saveAccountingService = function (saveReturn, sendFunction, statusExpected, done) {
+               var config = {
+                    appHost: DEFAULT_WSTOREHOST,
+               };
+               var req = {
+                    ip: {
+                         replace: function (expr) { return DEFAULT_WSTOREHOST}
+                    },
+                    body: '{ "url": "' + DEFAULT_URL + '"}'
+               };
+
+               var res = jasmine.createSpyObj('res', ['status', sendFunction]);
+               res.status.and.returnValue(res);
+
+               var serviceInstance = jasmine.createSpyObj('service', ['save']);
+               serviceInstance.save.and.callFake( function (callback) {
+                    return callback(saveReturn);
+               });
+               var accServiceSchema = jasmine.createSpy();
+               accServiceSchema.and.returnValue(serviceInstance);
+
+               var accSerivceController = getAuthorizeServiceController(accServiceSchema, config);
+
+               accSerivceController.getApiKey(req, res);
+
+               setTimeout(function () {
+                    expect(serviceInstance.url).toBe('http://example/path');
+                    expect(serviceInstance.state).toBe('UNCOMMITTED');
+
+                    expect(res.status).toHaveBeenCalledWith(statusExpected);
+                    expect(res[sendFunction]).toHaveBeenCalled();
+
+                    done();
+               }, 100);
+          };
+
+          it('should return 500 when db fails', function (done) {
+               saveAccountingService('Error', 'send', 500, done);
+          });
+
+          it('Should generate and save a new apiKey with "UNCOMMITTED" state', function (done) {
+               saveAccountingService(null, 'json', 201, done);
```

Comment:
In this case you should also check that the `json` function has been called with the appropriate parameters.